### PR TITLE
Digital voucher api stub

### PR DESCRIPTION
**Pull request by frj**
This creates a starting point for the digital voucher api.

Stubbed endpoints have been created that return dummy data to support developing the clients of the api.

The endpoints are defined in the following spreadsheet:

https://docs.google.com/spreadsheets/d/1j1mLqnMwmQl_V6ZU6VOvisnftKHC1UyNeNp4NdJ1gWo/edit#gid=0

### CODE REVIEW

File: handlers/digital-voucher-api/src/test/scala/com/gu/digital_voucher_api/DigitalVoucherApiTest.scala [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import org.http4s.{Method, Request, Response, Uri}
+import org.scalatest.{EitherValues, FlatSpec, Matchers}
+
+class DigitalVoucherApiTest extends FlatSpec with Matchers with EitherValues {
+  "DigitalVoucherApi" should "return stubbed voucher details for create request" in {
+    val app = createApp()
+    val response = app.run(
+      Request(
+        method = Method.PUT,
+        Uri(path = s"/digital-voucher/create/sub123456")
```
Comment (kelvin-chappell) [inline handlers/digital-voucher-api/src/test/scala/com/gu/digital_voucher_api/DigitalVoucherApiTest.scala]:
```suggestion
        Uri(path = "/digital-voucher/create/sub123456")
```

File: handlers/digital-voucher-api/src/main/scala/com/gu/digital_voucher_api/DigitalVoucherService.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      ratePlanName: String
+    ): EitherT[F, DigitalVoucherServiceError, Voucher] =
+      EitherT.rightT[F, DigitalVoucherServiceError](Voucher(s"$subscriptionId-card-code", s"$subscriptionId-letter-code"))
+
+    override def replaceVoucher(
+      subscriptionId: String,
+      ratePlanName: String
+    ): EitherT[F, DigitalVoucherServiceError, Voucher] =
+      EitherT.rightT[F, DigitalVoucherServiceError](
+        Voucher(s"$subscriptionId-replaced-card-code", s"$subscriptionId-replaced-letter-code")
```
Comment (kelvin-chappell) [inline handlers/digital-voucher-api/src/main/scala/com/gu/digital_voucher_api/DigitalVoucherService.scala]:
```suggestion
        Voucher(subscriptionId-replaced-card-code, subscriptionId-replaced-letter-code)
```

File: handlers/digital-voucher-api/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+Digital Voucher Api
+====================
+
+This api provides access to Imovos digital voucher services. This supports the provisioning and
```
Comment (kelvin-chappell) [inline handlers/digital-voucher-api/README.md]:
```suggestion
This api provides access to Imovo's digital voucher services. This supports the provisioning and 
```

File: handlers/digital-voucher-api/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+Digital Voucher Api
```
Comment (kelvin-chappell) [inline handlers/digital-voucher-api/README.md]:
```suggestion
Digital Voucher API
```

File: handlers/digital-voucher-api/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+=====
+
+All endpoints require...
+
+- `x-api-key` header (generated by the `DigitalVoucherApiKey` in the CloudFormation)
+
+| Method | Endpoint | Body | Response | Description |
+| --- | --- | --- | --- | --- |
+| GET | `/{STAGE}/digital-voucher/\<SALESFORCE_SUBSCRIPTION_ID\> | | {"cardCode":"\<Imovo Card Code\>","letterCode":"\<Imovo Letter Code\>"} | Returns the Imovo digital subscription associated with the salesforce subscription id |
+| PUT | `/{STAGE}/digital-voucher/create/\<SALESFORCE_SUBSCRIPTION_ID\> | {"ratePlanName":"\<subscription rate plan name\>"} | {"cardCode":"\<Imovo Card Code\>","letterCode":"\<Imovo Letter Code\>"} | Creates an Imovo digital subscription or returns the details the subscription if it already exists |
```
Comment (kelvin-chappell) [inline handlers/digital-voucher-api/README.md]:
```suggestion
| PUT | `/{STAGE}/digital-voucher/create/\<SALESFORCE_SUBSCRIPTION_ID\> | {"ratePlanName":"\<subscription rate plan name\>"} | {"cardCode":"\<Imovo Card Code\>","letterCode":"\<Imovo Letter Code\>"} | Creates an Imovo digital subscription or returns the details of the subscription if it already exists |
```

**Comment by arausuy** [comment]
Great work 👍

File: handlers/digital-voucher-api/src/main/scala/com/gu/digital_voucher_api/DigitalVoucherService.scala (hunk 2 above)
Comment (frj) [reply handlers/digital-voucher-api/src/main/scala/com/gu/digital_voucher_api/DigitalVoucherService.scala]: I think these ones are legit, the idea is to use the sub id from the request to generate the values returned by the api.

**Pull request merged**